### PR TITLE
API-7198 Terms of Use page to use status from Terms of Use invitation record

### PR DESCRIPTION
**Pull request by petekirby-ee**

### CODE REVIEW

File: app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/TermsOfUseHistoryController.scala [hunk 1]
```diff
@@ -93,6 +102,14 @@ class TermsOfUseHistoryController @Inject() (
       }
     }
 
+    def deriveInvitationStatusDescription(status: TermsOfUseInvitationState): String = {
+      status match {
+        case REMINDER_EMAIL_SENT => "We sent a reminder email to the admins of the application."
+        case OVERDUE             => "The terms of use has not been completed by the due date."
```
Comment (johnsgp) [inline app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/TermsOfUseHistoryController.scala:108]: Grammar looks wrong. "terms" is plural, so should be "have" not "has" ?
Alternatively, "The terms of use acceptance has not been ...."

File: app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/TermsOfUseHistoryController.scala [hunk 2]
```diff
@@ -117,16 +134,56 @@ class TermsOfUseHistoryController @Inject() (
...
-        "Email sent",
-        "We invited admins of the application to agree to version 2 of the terms of use.",
+        deriveInvitationStatusDisplayName(EMAIL_SENT),
+        deriveInvitationStatusDescription(EMAIL_SENT),
+        None,
+        None
+      )
+    }
+
+    def buildReminderEmailSentModelFromInvitation(invite: TermsOfUseInvitation): TermsOfUseHistory = {
```
Comment (johnsgp) [inline app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/TermsOfUseHistoryController.scala:147]: Feels like some deduplication of code could happen here by passing in the date/time? I may be wrong.

**Review by Tobias1087 — APPROVED** [review]

File: app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/views/TermsOfUsePage.scala.html [hunk 3]
```diff
@@ -23,14 +23,14 @@
...
 
 @buildStatus(status: String) = @{
-  status.toUpperCase() match {
-    case "EMAIL SENT"                     => <strong class="govuk-tag govuk-tag--blue">{status}</strong>
-    case "IN PROGRESS"                    => <strong class="govuk-tag govuk-tag--blue">{status}</strong>
-    case "PENDING RESPONSIBLE INDIVIDUAL" => <strong class="govuk-tag govuk-tag--blue">{status}</strong>
+  status match {
+    case "EMAIL_SENT"                     => <strong class="govuk-tag govuk-tag--blue">EMAIL SENT</strong>
+    case "REMINDER_EMAIL_SENT"            => <strong class="govuk-tag govuk-tag--blue">REMINDER EMAIL SENT</strong>
+    case "OVERDUE"                        => <strong class="govuk-tag govuk-tag--blue">{status}</strong>
```
Comment (johnsgp) [inline app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/views/TermsOfUsePage.scala.html:29]: Are "IN PROGRESS" and "PENDING RESPONSIBLE INDIVIDUAL" supposed to be removed here?

File: app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/TermsOfUseHistoryController.scala [hunk 4]
```diff
@@ -117,16 +134,56 @@ class TermsOfUseHistoryController @Inject() (
...
+      TermsOfUseHistory(
+        invite.reminderSent.fold("Unknown")(r => DateTimeFormatter.ofPattern("dd MMMM yyyy").withZone(ZoneId.systemDefault()).format(r)),
```
Comment (johnsgp) [inline app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/TermsOfUseHistoryController.scala:149]: I can't find a test for this fold.

File: app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/views/TermsOfUsePage.scala.html (hunk 3 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/views/TermsOfUsePage.scala.html:29]: Yes, that's deliberate

File: app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/TermsOfUseHistoryController.scala (hunk 1 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/TermsOfUseHistoryController.scala]: Fixed

File: app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/TermsOfUseHistoryController.scala (hunk 2 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/TermsOfUseHistoryController.scala]: Yep, I agree - fixed.

File: app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/TermsOfUseHistoryController.scala (hunk 4 above)
Comment (petekirby-ee) [reply app/uk/gov/hmrc/apigatekeeperapprovalsfrontend/controllers/TermsOfUseHistoryController.scala]: Fixed

**Pull request merged**